src/Control/Optics/Linear/Traversal.hs

```diff
@@ -1,3 +1,53 @@
+{-# LANGUAGE LinearTypes #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE GADTs #-}
+-- | This module provides linear traversals
+--
+--  Traversals provides a means of accessing several @a@s organized in some
+--  structural way in an @s@, and a means of changing them to @b@s to create a
+--  @t@. In very ordinary language, it's like walking or traversing the data
+--  structure, going across cases and inside definitions. In more imaganitive
+--  language, it's like selecting some specific @a@s by looking at each
+--  constructor of a data definition and recursing on each non-base type.
+--
+-- = Example
+--
+-- @
+-- import Prelude (String, Int)
+-- import Prelude.Linear ((&))
+-- import qualified Control.Monad.Linear as Control
+-- import Data.List.Linear (traverse', (++))
+-- import Control.Monad.Linear ((<$>), (<*>))
+--
+-- -- We can use a traversal to append a string only to the 
+-- -- human names in a classroom struct
+-- appendToNames :: String -> Classroom %1-> Classroom
+-- appendToNames s = over classroomNamesTrav (\name -> name ++ s)
+--
+-- data Classroom where
+--   Classroom ::
+--     { className :: String
+--     , teacherName :: String
+--     , classNum :: Int
+--     , students :: [Student]
+--     , textbooks :: [String]
+--     } %1-> Classroom
+--
+-- -- A Student is a name and a student id number
+-- data Student = Student String Int
+--
+-- classroomNamesTrav :: Traversal' Classroom String
+-- classroomNamesTrav = traversal traverseClassStr where
+--   traverseClassStr :: forall f. Control.Applicative f =>
+--     (String %1-> f String) -> Classroom %1-> f Classroom
+--   traverseClassStr onName classroom = classroom &
+--     \(Classroom cname teachname x students texts) ->
```

Comment:
Isn't this exactly the same as pattern matching on `classroom` right at the function bindings:

```haskell
-   traverseClassStr onName (Classroom cname teachname x students texts) = ...
```